### PR TITLE
Convert action-based logic to intent-based logic

### PR DESCRIPTION
**Pull request by CyrusJia**
Remove onActionNextStep() from app.js and events.js

Modify onAction to onIntent and adjust corresponding logic

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# watsonwork-weather
+# watsonwork-weather-intent-based
```
Comment (jsdelfino) [inline README.md]: We're not changing the name of this project are we?

File: src/events.js [hunk 2]
```diff
@@ -11,55 +11,38 @@ const log = debug('watsonwork-weather-events');
 // event, the original annotated message and the user who sent it
 const callback = (evt, appId, info, annotation, token, cb) => {
 
-  // Retrieve the annotated message
+    // Retrieve the annotated message
```
Comment (jsdelfino) [inline src/events.js]: Not sure why the formatting changes here (we've been using spaces for indentation, is this a tab?)

File: README.md (hunk 1 above)
Comment (CyrusJia) [reply README.md]: Would it make sense to separate this project into two different repos: one for action responses and one for intent responses?

File: src/events.js [hunk 3]
```diff
@@ -11,55 +11,38 @@ const log = debug('watsonwork-weather-events');
...
   messages.message(evt.messageId, token(), (err, message) => {
-    if(err)
+    if (err)
       return;
 
-    // Ignore messages from the app itself
-    if(message.createdBy.id === appId)
+        // Ignore messages from the app itself
+    if (message.createdBy.id === appId)
```
Comment (jsdelfino) [inline src/events.js]: Conflicts with the current coding style of this example app (no space between if and ()

File: src/events.js [hunk 4]
```diff
@@ -11,55 +11,38 @@ const log = debug('watsonwork-weather-events');
...
       return;
 
-    // Return the extracted info, annotation, annotated message
-    // and the user who sent it
+        // Return the extracted info, annotation, annotated message
```
Comment (jsdelfino) [inline src/events.js]: Looks like another unnecessary formatting change (more of these in this pull request)

File: src/events.js [hunk 5]
```diff
@@ -11,55 +11,38 @@ const log = debug('watsonwork-weather-events');
...
+          util.inspect(message, { colors: debug.useColors(), depth: 10 }));
     cb(info, annotation, message, message.createdBy);
   });
 };
 
 // Return the action identified in an annotation event
-export const onAction = (evt, appId, token, cb) => {
+export const onIntent = (evt, appId, token, cb) => {
+  // log('Annotation event!');
+  log(evt);
```
Comment (jsdelfino) [inline src/events.js]: Would be good to have a log message too here

File: src/app.js [hunk 6]
```diff
@@ -1,6 +1,16 @@
 // A sample app that listens to messages posted to a space in IBM
```
Comment (jsdelfino) [inline src/app.js]: Many many changes here probably caused by your editor. Can you please re-apply the minimum set of changes that will do what you intended to do? and I'll be happy to review again :) Thanks!

Comment (CyrusJia) [reply src/app.js]: Sure, no problem!

File: src/app.js [hunk 7]
```diff
@@ -37,54 +37,23 @@ export const weather = (appId, store, wuser, wpassword, token) =>
     // be sent asynchronously
     res.status(201).end();
 
-    // Handle messages identified as action requests
-    events.onAction(req.body, appId, token,
-      (action, focus, message, user) => {
+    // Handle messages identified as intent requests
```
Comment (jsdelfino) [inline src/app.js]: I'd suggest to rephrase as something like 'messages on which an intent has been recognized'

Comment (jsdelfino) [reply src/app.js]: or 'identified' in place of 'recognized'

File: src/app.js [hunk 8]
```diff
@@ -136,46 +105,68 @@ export const weather = (appId, store, wuser, wpassword, token) =>
...
-              // weather forecast in the recognized city
-              if(astate.action === 'Get_Weather_Conditions')
-                send(confirmConditions(city, user));
+            // Ask the user to confirm
+            if(intent === 'weather')
+              send(confirmConditions(city, user));
 
-              else if(astate.action === 'Get_Weather_Forecast')
-                send(confirmForecast(city, user));
+            else if(intent == 'forecast')
```
Comment (jsdelfino) [inline src/app.js]: I'm surprised that Lint didn't pick that up, but I'd suggest to change == to ===

**Review by jsdelfino — CHANGES_REQUESTED** [review]
Looks pretty good to me, I added a few comments for some minor issues.

Can you convert it to a pull request with a single clean commit containing an overview of the change that's implemented here?

Also I'm guessing that you didn't mean to push your .DS_Store file :)

With these minor changes I'll be +1 on merging this pull request. Thanks again for your contribution!!

**Comment by jsdelfino** [comment]
Thanks for this contribution!

Closing this pull request as the code changes have now been integrated in pull request #4.

**Pull request closed without merging**